include/dlaf/common/index2d.h

```diff
@@ -145,6 +149,10 @@ class Size2D : public internal::basic_coords<IndexT> {
 public:
   using BaseT::basic_coords;
 
+  explicit operator Index2D<IndexT, Tag>() const noexcept {
+    return {rows(), cols()};
+  }
```

Comment:
I would use a method instead of conversion, such that the operation done is eplicitly stated:
e.g.:
`Index indexFromOrigin(size)` i.e. index{0, 0} + size
Moreover I would add also the inverse operation:
`Size sizeFromOrigin(index)` i.e index - Index{0, 0}

Reply:
I followed your suggestion. I agree that the cast operator, even if explicitly required, it is not so self-explicative.

About the name of these functions, I'm not really sure. I want to make sure that the user does not interpret it wrongly (e.g. it has to pass an origin?! it doesn't make much sense).

Please consider something slightly more verbose like `indexWrtOriginOf(size)`.

Reply:
The names were just a suggestion... if you have better options feel free to use them.